mmv1/third_party/terraform/services/resourcemanager/ephemeral_google_service_account_token.go

```diff
@@ -0,0 +1,143 @@
+package resourcemanager
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/ephemeral"
+	"github.com/hashicorp/terraform-plugin-framework/ephemeral/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-provider-google/google/fwtransport"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	"google.golang.org/api/iamcredentials/v1"
+)
+
+var _ ephemeral.EphemeralResource = &googleEphemeralServiceAccountAccessToken{}
+
+func GoogleEphemeralServiceAccountAccessToken() ephemeral.EphemeralResource {
+	return &googleEphemeralServiceAccountAccessToken{}
+}
+
+type googleEphemeralServiceAccountAccessToken struct {
+	providerConfig *fwtransport.FrameworkProviderConfig
+}
+
+func (p *googleEphemeralServiceAccountAccessToken) Metadata(ctx context.Context, req ephemeral.MetadataRequest, resp *ephemeral.MetadataResponse) {
+	resp.TypeName = "google_test"
+}
+
+type ephemeralServiceAccountAccessTokenModel struct {
+	TargetServiceAccount types.String `tfsdk:"target_service_account"`
+	AccessToken          types.String `tfsdk:"access_token"`
+	Scopes               types.Set    `tfsdk:"scopes"`
+	Delegates            types.Set    `tfsdk:"delegates"`
+	Lifetime             types.String `tfsdk:"lifetime"`
+}
+
+func (p *googleEphemeralServiceAccountAccessToken) Schema(ctx context.Context, req ephemeral.SchemaRequest, resp *ephemeral.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"target_service_account": schema.StringAttribute{
+				Required: true,
+				// Validators: verify.ValidateRegexp("(" + strings.Join(verify.PossibleServiceAccountNames, "|") + ")"),
+			},
+			"access_token": schema.StringAttribute{
+				Sensitive: true,
+				Computed:  true,
+			},
+			"lifetime": schema.StringAttribute{
+				Optional: true,
+				// Validators: verify.ValidateDuration(), // duration <=3600s; TODO: support validateDuration(min,max)
```

Comment:
Note: see here for existing file for validators (+tests) : https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/fwprovider/framework_validators.go

Reply:
The new validators can be added into `framework_validators.go`

https://github.com/GoogleCloudPlatform/magic-modules/pull/12140/files#diff-ead6dccc32e7d20e09634bc1acac892f524984224bb21149b27d1a64f548b912R154-R233

Reply:
when attempting to move the new validators over to `framework_validators.go` we get an `import cycle` error.

This is due to `fwprovider` importing `resourcemanager` in order to include support for ephemeral resources (as well as plugin data sources previously included.)

A separate PR needs to be made where we create `fwvalidators` in order to resolve this: https://github.com/GoogleCloudPlatform/magic-modules/pull/12268

Reply:
Thanks for explaining- the import cycle restriction makes sense and solving it in a separate PR is appropriate